src/components/ArticlesList.jsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
+import { getArticles } from "../api";
+import ArticleCard from "./ArticleCard";
+
+export default function ArticlesList() {
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getArticles()
+      .then((articles) => {
+        console.log(articles);
+        setArticles(articles);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        setError(true);
+        console.error("Error fetching articles:", error);
+        setIsLoading(false);
+      });
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
+
+  if (error) {
+    return <p>Error fetching articles data: {error.message}</p>;
+  }
+
+  return (
+    <div class="articlesList">
```

Comment:
you can't use `class` in HTMX for HTML classes. You need to use `className` (or it will create errors like the one I'm reading in your browser):

Warning: Invalid DOM property `class`. Did you mean `className`?
    at header
    at Header
    at section
    at App
    at Router (https://5173-sabzm-ncnewsfe-rs1d389v615.ws-eu117.gitpod.io/node_modules/.vite/deps/react-router.js?v=2a776e43:6030:13)